docs/mllib-regression.md

```diff
@@ -0,0 +1,161 @@
+---
+layout: global
+title: Naive Bayes - MLlib
+displayTitle: <a href="mllib-guide.html">MLlib</a> - Regression
+---
+
+## Regression
+[Regression](http://en.wikipedia.org/wiki/Regression_analysis) is a statistical process
+for estimating the relationships among variables. It includes many techniques for modeling
+and analyzing several variables, when the focus is on the relationship between
+a dependent variable and one or more independent variables.
+
+## Isotonic regression
+[Isotonic regression](http://en.wikipedia.org/wiki/Isotonic_regression)
+belongs to the family of regression algorithms. Formally isotonic regression is a problem where
+given a finite set of real numbers `$Y = {y_1, y_2, ..., y_n}$` representing observed responses
+and `$X = {x_1, x_2, ..., x_n}$` the unknown response values to be fitted
+finding a function that minimises
+
+`\begin{equation}
+  f(x) = \sum_{i=1}^n w_i (y_i - x_i)^2
+\end{equation}`
+
+with respect to complete order subject to
+`$x_1\le x_2\le ...\le x_n$` where `$w_i$` are positive weights.
+The resulting function is called isotonic regression and it is unique.
+It can be viewed as least squares problem under order restriction.
+Essentially isotonic regression is a
+[monotonic function](http://en.wikipedia.org/wiki/Monotonic_function)
+best fitting the original data points.
+
+MLlib supports a
+[pool adjacent violators algorithm](http://www.stat.cmu.edu/~ryantibs/papers/neariso.pdf)
```

Comment:
Use the DOI link http://doi.org/10.1198/TECH.2010.10111 instead, which probably lives longer.